sdk/core/core/test/test_json_pointer_parser.h

```diff
@@ -11,50 +11,151 @@ static void test_json_pointer_parser() {
   {
     az_span_reader parser = az_span_reader_create(AZ_STR(""));
     az_span p;
-    TEST_ASSERT(az_json_pointer_parser_get(&parser, &p) == AZ_ERROR_ITEM_NOT_FOUND);
+    TEST_ASSERT(az_span_reader_read_json_pointer_token(&parser, &p) == AZ_ERROR_ITEM_NOT_FOUND);
   }
   {
     az_span_reader parser = az_span_reader_create(AZ_STR("Hello"));
     az_span p;
-    TEST_ASSERT(az_json_pointer_parser_get(&parser, &p) == AZ_ERROR_PARSER_UNEXPECTED_CHAR);
+    TEST_ASSERT(
+        az_span_reader_read_json_pointer_token(&parser, &p) == AZ_ERROR_PARSER_UNEXPECTED_CHAR);
   }
   {
     az_span_reader parser = az_span_reader_create(AZ_STR("/abc"));
     az_span p;
-    TEST_ASSERT(az_json_pointer_parser_get(&parser, &p) == AZ_OK);
+    TEST_ASSERT(az_span_reader_read_json_pointer_token(&parser, &p) == AZ_OK);
     TEST_ASSERT(az_span_eq(p, AZ_STR("abc")));
-    TEST_ASSERT(az_json_pointer_parser_get(&parser, &p) == AZ_ERROR_ITEM_NOT_FOUND);
+    // test az_json_pointer_token_parser_get
+    {
+      az_span_reader token_parser = az_span_reader_create(p);
+      uint8_t buffer[10];
+      int i = 0;
+      while (true) {
+        az_result const result
+            = az_span_reader_read_json_pointer_token_char(&token_parser, buffer + i);
+        if (result == AZ_ERROR_ITEM_NOT_FOUND) {
```

Comment:
what if it ever throw a different ERROR?  don't we want to validate that to avoid unknown behavior here?

Reply:
Next line `TEST_ASSERT(result == AZ_OK);`